src/lib/managers/StoreManager.ts

```diff
@@ -0,0 +1,31 @@
+import { StoreAPI } from '@/lib/api';
+import { getClientCookie } from '@/lib/services/CookieService';
+import type { UUID } from '@/lib/types';
+import type { PublicOrderWithItems } from '@/lib/types/apiResponses';
+import type { ClientCartItem } from '@/lib/types/client';
+import { CookieType } from '@/lib/types/enums';
+
+//* disabling this until more functions are added
+/* eslint-disable import/prefer-default-export */
+
+/**
+ * Place a new merch order
+ * @param items array of ClientCartItem to order
+ * @param pickupEvent uuid of a pickup event
+ * @returns successfully created order
+ */
+export const placeMerchOrder = (items: ClientCartItem[], pickupEvent: UUID) =>
+  new Promise<PublicOrderWithItems>((resolve, reject) => {
+    const token = getClientCookie(CookieType.ACCESS_TOKEN);
+    if (!token) reject(new Error('Missing access token'));
+
+    StoreAPI.placeMerchOrder(token, {
+      order: items.map(({ option: { uuid }, quantity }) => ({
+        option: uuid,
+        quantity,
+      })),
+      pickupEvent,
+    })
+      .then(res => resolve(res))
+      .catch(err => reject(err.response.data.error));
+  });
```

Comment:
You should just convert this to an async function

```suggestion
export const placeMerchOrder = async (items: ClientCartItem[], pickupEvent: UUID) =>
  const token = getClientCookie(CookieType.ACCESS_TOKEN);
  if (!token) throw new Error('Missing access token');

  return StoreAPI.placeMerchOrder(token, {
    order: items.map(({ option: { uuid }, quantity }) => ({
      option: uuid,
      quantity,
    })),
    pickupEvent,
  });
}
```

then use `reportError` from lib/utils to handle the error (since `err.response.data.error` might itself cause an error if there was a network error rather than api error)

Reply:
the new getErrorMessage function is awesome!